sc/social/like/browser/canonicalurl.py

```diff
@@ -106,9 +107,7 @@ def update_canonical_url(self, data):
         logger.info(u'{0} objects will have their canonical URL updated'.format(total))
 
         for obj in get_valid_objects(results):
-            # FIXME: we're currently ignoring the Plone site id
-            #        https://github.com/collective/sc.social.like/issues/119
-            path = '/'.join(obj.getPhysicalPath()[2:])
+            path = get_path_to_virtual_path(self.request, obj)
```

Comment:
Yeah, that's tough. I can't see a safer way to get the previous URL than accessing it with it (eg. with VHM).

However, asking the user to choose to use or not the _Site Id_ as you said, sounds better than ask for the old _VirtualHostMonster_ URL. :thinking: 

Is this case, we should allow path in the `canonical_url`. Imagine `http://localhost:8080/VirtualHostBase/http/foo.com:80/Plone/VirtualHostRoot/_vh_intranet`
Just saying to not use the _Site Id_ isn't enough, you have to say that the root URL was `http://foo.com/intranet/`